test/e2e/performanceprofile/functests/1_performance/cpu_management.go

```diff
@@ -671,12 +671,13 @@ var _ = Describe("[rfe_id:27363][performance] CPU Management", Ordered, func() {
 						Expect(pod.Status.QOSClass).To(Equal(corev1.PodQOSGuaranteed))
 						if pod.Status.Phase != corev1.PodRunning && !isPodReady(&pod) {
 							for _, containerStatus := range pod.Status.ContainerStatuses {
-								if containerStatus.State.Waiting != nil {
-									if containerStatus.State.Waiting.Reason == "RunContainerError" {
-										if strings.Contains(containerStatus.State.Waiting.Message, "failed to run pre-start hook for container") {
-											testlog.Infof("container %s failed to start with error: %s", pod.Spec.Containers[0].Name, containerStatus.State.Waiting.Message)
-											return false
-										}
+								if containerStatus.State.Waiting == nil {
+									continue
+								}
+								if containerStatus.State.Waiting.Reason == "RunContainerError" {
+									if strings.Contains(containerStatus.State.Waiting.Message, "failed to run pre-start hook for container") {
+										testlog.Infof("container %s failed to start with error: %s", pod.Spec.Containers[0].Name, containerStatus.State.Waiting.Message)
+										return false
 									}
```

Comment:
if the code doesn't return false here the test will pass but not because the deployment is ready in all instances but because the test ignored the pod that is in different waiting reason. this will be misleading, I'm not sure this is what we want, is it? 
to have a negative check, maybe increase the requested cpus in such case that the node can't provide resources (not enough resources), and verify if the test pass, if it does then it is false negative

Reply:
Added additional checks to verify deployment is indeed successful